doc/md/fundamentals/11-error-handling.md

```diff
@@ -0,0 +1,99 @@
+---
+sidebar_position: 11
+---
+
+# Error handling
+
+| Method | Purpose | Behavior | Example use case |
+|--------|---------|----------|----------|
+| Error | Represents recoverable errors using `throw` and `try/catch`. | Can be caught and handled. | Handling messaging errors. |
+| Option (`?T`) | Represents a value that may be missing. | Returns `null` if no value is available. | Searching for an item, optional parameters. |
+| Result (`Result.Result<T, E>`) | Represents success (`#ok`) or failure (`#err`). | Encodes both valid results and errors in a structured format. | Operations that can fail, such as authentication. |
+
+In addition to explicit error handling, Motoko provides traps and assertions for dealing with execution errors.
+
+| Method | Purpose | Behavior | Example use case |
+|--------|---------|----------|----------|
+| Trap (`Debug.trap`) | Indicates an unrecoverable error. | Immediately stops execution and rolls back state. | Preventing invalid operations, handling fatal errors. |
+| Assertion (`assert`) | Ensures that a condition holds. | Traps if the condition is false. | Enforcing invariants, validating input. |
+
+## Traps
+
+Traps immediately stop execution and roll back [state](https://internetcomputer.org/docs/motoko/fundamentals/state). They are used for fatal errors that cannot be recovered.
+
+```motoko no-repl
+import Debug "mo:base/Debug";
+
+func divide(a : Nat, b : Nat) : Nat {
+    if (b == 0) {
+        Debug.trap("Cannot divide by zero");
+    };
+    return a / b;
+};
+```
+
+## Assertions
```

Comment:
And here add all the info about `debug assert` and `--release` `moc` flag and `Release` option from `dfx`.
See the comment in the style guide -- add a link to this section there